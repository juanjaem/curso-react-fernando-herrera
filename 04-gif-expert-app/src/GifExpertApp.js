import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'
import PropTypes from 'prop-types'

function GifExpertApp ({ defaultCategories = [] }) {
    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h1>GifExpertApp</h1>
            <hr></hr>

            <AddCategory setCategories={setCategories}/>

            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}

export default GifExpertApp
