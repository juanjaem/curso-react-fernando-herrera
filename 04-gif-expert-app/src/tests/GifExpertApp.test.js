import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp'

describe('pruebas en <GifExpertApp />', () => {
    test('debe de mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorÃ­as', () => {
        const categories = ['One Punch', 'Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});
