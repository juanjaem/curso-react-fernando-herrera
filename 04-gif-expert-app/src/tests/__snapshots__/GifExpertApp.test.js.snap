// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifExpertApp /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
